package/windows/hyperkube.ps1

```diff
@@ -488,6 +460,105 @@ function config-azure-cloudprovider {
     popd
 }
 
+function print-system-info {
+    print  "******"
+    print ("*****       CNI - Component: {0}, Mode: {1}" -f $KubeCNIComponent, $KubeCNIMode)
+    if ($NodeIP) {
+        print ("****       Node - Name: {0}, IP: {1}, InternalIP: {2}" -f $NodeName, $NodePublicIP, $NodeIP)
+    } else {
+        print ("****       Node - Name: {0}, IP: {1}" -f $NodeName, $NodePublicIP)
+    }
+    print ("***     Cluster - CIDR: {0}, Domain: {1}, ServiceCIDR: {2}, DnsServiceIP: {3}" -f $KubeClusterCIDR, $KubeClusterDomain, $KubeServiceCIDR, $KubeDnsServiceIP)
+    print ("** ControlPlane - Host: {0}" -f $KubeControlPlaneAddresses)
+    if ($KubeletCloudProviderName) {
+        print ("* CloudProvider - Name: {0}" -f $KubeletCloudProviderName)
+    }
+    print "*"
+}
+
+function setup-proxy {
+    param(
+        [parameter(Mandatory = $true)] [string]$CPHosts
+    )
+
+    $tmpl =
+@'
+error_log /var/log/nginx.log notice;
+pid       /var/log/nginx.pid;
+worker_rlimit_nofile 8192;
+worker_processes auto;
+events {
+  multi_accept on;
+  worker_connections 4096;
+}
+
+stream {
+    upstream kube_apiserver {
+        least_conn;
+        {0}
+    }
+
+    server {
+        listen        127.0.0.1:6443 so_keepalive=on;
+        access_log    off;
+        proxy_pass    kube_apiserver;
+        proxy_timeout 10m;
+        proxy_connect_timeout 1s;
+    }
+}
+'@
+    $configPath = "$NginxConfigDir\nginx.conf"
+    $servers = ""
+    $tempConfig = New-TemporaryFile
+
+    $CPHosts -split "," | % {
+        $servers += ("server $_" + ":6443;`t")
+    }
+    $newConfig = $tmpl.Replace("{0}", $servers).ToString()
+
+    $mustReload = $False
+    if (Test-Path $configPath) {
+        $newConfigStream = [IO.MemoryStream]::new([Text.Encoding]::ASCII.GetBytes($newConfig))
+        $newConfigHasher = Get-FileHash -InputStream $newConfigStream -Algorithm sha256
+        $staleConfigHasher = Get-FileHash $configPath -Algorithm sha256
+        if ($staleConfigHasher.Hash -ne $newConfigHasher.Hash) {
+            $newConfig | Out-File -NoNewline -Encoding ascii -Force -FilePath $configPath
+            $mustReload = $True
+        } else {
+            $tempConfig.Delete()
+        }
+    } else {
+        $newConfig | Out-File -NoNewline -Encoding ascii -Force -FilePath $configPath
+    }
+
+    $process = Get-Process -Name "nginx" -ErrorAction Ignore
+    if ($process) {
+        if ($mustReload) {
+            print-system-info
+
+            print "Reloading controlplanes proxy ."
+            $process = Start-Process -PassThru -FilePath "$NginxConfigDir\nginx.exe" -ArgumentList "-c $configPath -s reload" -WindowStyle Hidden
+            Start-Sleep -s 5
+            $process = Get-Process -Id $process.Id -ErrorAction Ignore
+            if (-not $process) {
+                throw "........................ FAILED, agent retry"
+            }
+            print "........................... OK"
```

Comment:
whats up with the long line of dots? why do we have them?

Reply:
The dot dot dot line is used to indicate the Windows host setup progress, it keeps the same width to make the log looks like this:
![image](https://user-images.githubusercontent.com/3394724/57001431-3d36d100-6beb-11e9-8689-ca84b5df6bd0.png)